src/Mvc/Mvc.Core/src/Infrastructure/ProblemDetailsJsonConverter.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc.Core;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    internal class ProblemDetailsJsonConverter : JsonConverter<ProblemDetails>
+    {
+        private static readonly JsonEncodedText Type = JsonEncodedText.Encode("type");
+        private static readonly JsonEncodedText Title = JsonEncodedText.Encode("title");
+        private static readonly JsonEncodedText Status = JsonEncodedText.Encode("status");
+        private static readonly JsonEncodedText Detail = JsonEncodedText.Encode("detail");
+        private static readonly JsonEncodedText Instance = JsonEncodedText.Encode("instance");
+
+        public override ProblemDetails Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var problemDetails = new ProblemDetails();
+
+            if (!reader.Read())
+            {
+                throw new InvalidDataException(Resources.UnexpectedJsonEnd);
+            }
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                ReadValue(ref reader, problemDetails, options);
+            }
+
+            return problemDetails;
+        }
+
+        public override void Write(Utf8JsonWriter writer, ProblemDetails value, JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+            WriteProblemDetails(writer, value, options);
+            writer.WriteEndObject();
+        }
+
+        internal static void ReadValue(ref Utf8JsonReader reader, ProblemDetails value, JsonSerializerOptions options)
+        {
+            if (TryReadStringProperty(ref reader, Type, out var propertyValue))
+            {
+                value.Type = propertyValue;
+            }
+            else if (TryReadStringProperty(ref reader, Title, out propertyValue))
+            {
+                value.Title = propertyValue;
+            }
+            else if (TryReadStringProperty(ref reader, Detail, out propertyValue))
+            {
+                value.Detail = propertyValue;
+            }
+            else if (TryReadStringProperty(ref reader, Instance, out propertyValue))
+            {
+                value.Instance = propertyValue;
+            }
+            else if (reader.ValueTextEquals(Status.EncodedUtf8Bytes))
+            {
+                if (!reader.Read())
+                {
+                    throw new InvalidDataException(Resources.UnexpectedJsonEnd);
+                }
+
+                if (reader.TokenType == JsonTokenType.Null)
+                {
+                    // Nothing to do here.
+                }
+                else if (reader.TokenType == JsonTokenType.Number)
```

Comment:
Are we OK with silently ignoring if TokenType is something else or do we need an `else throw`?